File: Daily Challenge/Java/Super Binary String 1.11.2025.java
import java.util.Scanner;

public class PrefixCheck {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String n = sc.next();
        int a = 0, b = 0, c = 0;
        int count0 = 0, count1 = 0;

        for (int i = 0; i < n.length(); i++) {
            if (n.charAt(i) == '0') count0++;
            else if (n.charAt(i) == '1') count1++;
        }

        for (int i = 0; i < n.length(); i++) {
            if (n.charAt(i) == '1') a++;
            else b++;
            if (a < b) {
                c = 1;
                break;
            }
        }

        if (count0 == count1 && c == 0)
            System.out.println("Yes");
        else
            System.out.println("No");

        sc.close();
    }
}

File: Daily Challenge/C++/Super Binary String 1.11.2025.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string n;
    cin >> n;
    int a = 0, b = 0, c = 0;
    int count0 = 0, count1 = 0;
    for (char ch : n) {
        if (ch == '0') count0++;
        else if (ch == '1') count1++;
    }
    for (char ch : n) {
        if (ch == '1') a++;
        else b++;
        if (a < b) {
            c = 1;
            break;
        }
    }
    if (count0 == count1 && c == 0)
        cout << "Yes
";
    else
        cout << "No
";

    return 0;
}